spring-cloud-gcp-autoconfigure/src/test/java/org/springframework/cloud/gcp/autoconfigure/secretmanager/it/SecretManagerIntegrationTests.java

```diff
@@ -103,13 +103,52 @@ public void testConfigurationDisabled() {
 				.getProperty("spring-cloud-gcp.secrets.my-secret")).isNull();
 	}
 
+	@Test
+	public void testVersion() {
+		this.context = new SpringApplicationBuilder()
+				.sources(GcpContextAutoConfiguration.class, GcpSecretManagerBootstrapConfiguration.class)
+				.web(WebApplicationType.NONE)
+				.properties("spring.cloud.gcp.secretmanager.bootstrap.enabled=true")
+				.properties("spring.cloud.gcp.secretmanager.bootstrap.version=2")
+				.run();
+
+		createSecret(TEST_SECRET_ID, "the secret data");
+		assertThat(secretExists(TEST_SECRET_ID, "1")).isTrue();
+		createSecret(TEST_SECRET_ID, "the secret data v2");
+		assertThat(secretExists(TEST_SECRET_ID, "2")).isTrue();
+		assertThat(secretExists(TEST_SECRET_ID, "latest")).isTrue();
+
+		byte[] byteArraySecret = this.context.getEnvironment().getProperty("my-secret", byte[].class);
+		assertThat(byteArraySecret).isEqualTo("the secret data v2.".getBytes());
+	}
+
+	@Test
+	public void testSecretsWithSpecificVersion() {
```

Comment:
Can we add another test where both `secretmanager.version` and `secretmanager.versions` are specified and verify that the second takes precedence?